mempool/reactor_test.go

```diff
@@ -333,6 +334,43 @@ func TestReactorTxSendersMultiNode(t *testing.T) {
 	require.Zero(t, len(firstReactor.txSenders))
 }
 
+func TestMempoolFIFOWithParallelCheckTx(t *testing.T) {
+	if os.Getenv("CI") != "" {
+		// We expect this test to fail, but do not want to block merges because of it.
+		t.Skip("Skipping testing in CI environment")
+	}
+
+	config := cfg.TestConfig()
+	reactors, _ := makeAndConnectReactors(config, 4)
+	defer func() {
+		for _, r := range reactors {
+			if err := r.Stop(); err != nil {
+				assert.NoError(t, err)
+			}
+		}
+	}()
+	for _, r := range reactors {
+		for _, peer := range r.Switch.Peers().List() {
+			peer.Set(types.PeerStateKey, peerState{1})
+		}
+	}
+
+	// Deliver the same sequence of transactions from multiple sources, in parallel.
+	txs := newUniqueTxs(20)
```

Comment:
```suggestion
	txs := newUniqueTxs(100)
```

Reply:
I need to increase the number of txs to see the error on my machine. With 20, I never see it. Starting with 100 I see it consistently.

Reply:
Even when running a sequence of 100 tests? As in 
`go test -race -v -count=100 -failfast  -run TestMempool     ./mempool/...`